src/pages/Dashboard.tsx

```diff
@@ -10,28 +10,49 @@ import {addFiledRequest, deleteFiledRequest, getFiledRequests} from "../actions/
 import {FiledRequest} from "@interfaces/filedRequest";
 import {DEMO_REQUESTS} from "../databases/filedRequests";
 import {DEMO_INSIGHTS} from "../databases/insights";
+import {addUserToGroup, getUser, removeUserFromGroup} from "../actions/userActions";
+import {TEST_USER, User} from "@interfaces/user";
 
 const Dashboard = () => {
+    const [currentViewedGroup, setCurrentViewedGroup] = useState<string>(TEST_GROUP);
+    const [currentUserGroups, setUserGroups] = useState<string[]>([TEST_GROUP]);
     const [displayedRequests, setDisplayedRequests] = useState<{[id: string]: FiledRequest}>(DEMO_REQUESTS);
     const [displayedInsights, setDisplayedInsights] = useState<{[id: string]: Insight}>(DEMO_INSIGHTS);
+    const [newGroupNameInput, setNewGroupNameInput] = useState<string>("");
 
     useEffect(() => {
         (async () => {
             try {
-                const storedRequests = await getFiledRequests(TEST_ORGANIZATION, TEST_GROUP);
-                setDisplayedRequests({...displayedRequests, ...storedRequests});
+                // TODO User should be fetched and persisted upon login
+                const user: User = await getUser(TEST_USER);
+                setUserGroups([TEST_GROUP].concat(user.groupsIds));
             } catch (error: unknown) {
-                console.log("Fetching requests broken very sad")
-            }
-            try {
-                const storedInsights = await getInsights(TEST_ORGANIZATION, TEST_GROUP);
-                setDisplayedInsights({...displayedInsights, ...storedInsights});
-            } catch (error: unknown) {
-                console.log("Fetching insights broken very sad")
+                console.log("Fetching user broken very sad")
             }
+            await loadRequestsAndInsights(currentUserGroups[0]);
         })();
     }, [])
 
+    const loadRequestsAndInsights = async (groupName: string) => {
+        try {
+            const storedRequests = await getFiledRequests(TEST_ORGANIZATION, groupName);
+            setDisplayedRequests({...DEMO_REQUESTS, ...storedRequests});
+        } catch (error: unknown) {
+            console.log("Fetching requests broken very sad")
+        }
+        try {
+            const storedInsights = await getInsights(TEST_ORGANIZATION, groupName);
+            setDisplayedInsights({...DEMO_INSIGHTS, ...storedInsights});
+        } catch (error: unknown) {
+            console.log("Fetching insights broken very sad")
+        }
+    }
+
+    const switchViewedGroup = async (groupName: string) => {
+        setCurrentViewedGroup(groupName);
+        await loadRequestsAndInsights(groupName);
+    }
```

Comment:
If you add currentViewedGroup into the dependency array in the above comment you can make this happen in sequence. Also you don't need to await the loadRequestsAndInsights